### ISSUE TITLE
On V13 Beta - Point of Sales is blank showing - IndexError: list index out of range

### ISSUE THREAD

**Issue by PatroxGaurab** [frappe/erpnext#23004]
## Context information (for bug reports)

**Output of `bench version`**
```
erpnext 13.0.0-beta.4
frappe 13.0.0-beta.4
```

## Steps to reproduce the issue

1. open Point of sale from search bar
2. You will get the error

### Stacktrace / full error message

```

```

## Additional information

OS version / distribution, `ERPNext` install method, etc.
Install method OVA file download for V12 and then switch branch to v13 and bench migrate

**Comment by PatroxGaurab** [comment]
![Screenshot from 2020-08-12 01-23-53](https://user-images.githubusercontent.com/3338871/89942612-84a76c00-dc3a-11ea-8adc-fd384bf7e310.png)

**Comment by olragon** [comment]
It's also happened to me with Docker's Single Bench installation method. Hope these stack trace will help.

Request 1
```
POST /api/method/erpnext.selling.page.point_of_sale.point_of_sale.get_items
BODY:
start: 0
page_length: 40
price_list: Standard Selling
item_group: All Item Groups
search_value: 
pos_profile: LOC AAA
```

Stack trace 1
```
["Traceback (most recent call last):
  File \"/home/frappe/frappe-bench/apps/frappe/frappe/app.py\", line 64, in application
    response = frappe.api.handle()
  File \"/home/frappe/frappe-bench/apps/frappe/frappe/api.py\", line 58, in handle
    return frappe.handler.handle()
  File \"/home/frappe/frappe-bench/apps/frappe/frappe/handler.py\", line 30, in handle
    data = execute_cmd(cmd)
  File \"/home/frappe/frappe-bench/apps/frappe/frappe/handler.py\", line 69, in execute_cmd
    return frappe.call(method, **frappe.form_dict)
  File \"/home/frappe/frappe-bench/apps/frappe/frappe/__init__.py\", line 1086, in call
    return fn(*args, **newargs)
  File \"/home/frappe/frappe-bench/apps/erpnext/erpnext/selling/page/point_of_sale/point_of_sale.py\", line 20, in get_items
    warehouse, display_items_in_stock = frappe.db.get_value('POS Profile', pos_profile, ['warehouse', 'display_items_in_stock'])
  File \"/home/frappe/frappe-bench/apps/frappe/frappe/database/database.py\", line 408, in get_value
    order_by, cache=cache)
  File \"/home/frappe/frappe-bench/apps/frappe/frappe/database/database.py\", line 452, in get_values
    out = self._get_values_from_table(fields, filters, doctype, as_dict, debug, order_by, update)
  File \"/home/frappe/frappe-bench/apps/frappe/frappe/database/database.py\", line 599, in _get_values_from_table
    as_dict=as_dict, debug=debug, update=update)
  File \"/home/frappe/frappe-bench/apps/frappe/frappe/database/database.py\", line 158, in sql
    self._cursor.execute(query, values)
  File \"/home/frappe/frappe-bench/env/lib/python3.7/site-packages/pymysql/cursors.py\", line 170, in execute
    result = self._query(query)
  File \"/home/frappe/frappe-bench/env/lib/python3.7/site-packages/pymysql/cursors.py\", line 328, in _query
    conn.query(q)
  File \"/home/frappe/frappe-bench/env/lib/python3.7/site-packages/pymysql/connections.py\", line 517, in query
    self._affected_rows = self._read_query_result(unbuffered=unbuffered)
  File \"/home/frappe/frappe-bench/env/lib/python3.7/site-packages/pymysql/connections.py\", line 732, in _read_query_result
    result.read()
  File \"/home/frappe/frappe-bench/env/lib/python3.7/site-packages/pymysql/connections.py\", line 1075, in read
    first_packet = self.connection._read_packet()
  File \"/home/frappe/frappe-bench/env/lib/python3.7/site-packages/pymysql/connections.py\", line 684, in _read_packet
    packet.check_error()
  File \"/home/frappe/frappe-bench/env/lib/python3.7/site-packages/pymysql/protocol.py\", line 220, in check_error
    err.raise_mysql_exception(self._data)
  File \"/home/frappe/frappe-bench/env/lib/python3.7/site-packages/pymysql/err.py\", line 109, in raise_mysql_exception
    raise errorclass(errno, errval)
pymysql.err.InternalError: (1054, \"Unknown column 'display_items_in_stock' in 'field list'\")
```


Request 2
```
POST /api/method/runserverobj
BODY:
docs: {"docstatus":0,"doctype":"POS Invoice","name":"New POS Invoice 1","__islocal":1,"__unsaved":1,"owner":"my-email@gmail.com","naming_series":"ACC-PSINV-.YYYY.-","is_pos":1,"is_return":0,"company":"COMPANY XXX","posting_date":"2020-08-11","posting_time":"19:55:20","set_posting_time":0,"update_billed_amount_in_sales_order":0,"territory":"All Territories","currency":"VND","selling_price_list":"Standard Selling","price_list_currency":"VND","ignore_pricing_rule":0,"update_stock":0,"total_billing_amount":0,"redeem_loyalty_points":0,"apply_discount_on":"Grand Total","allocate_advances_automatically":0,"write_off_outstanding_amount_automatically":0,"group_same_items":0,"customer_group":"All Customer Groups","is_discounted":0,"status":"Draft","party_account_currency":"VND","is_opening":"No","c_form_applicable":"No","items":[],"__run_link_triggers":1,"pos_profile":"LOC AAA"}
method: set_missing_values
```

Stack trace 2
```
["Traceback (most recent call last):
  File \"/home/frappe/frappe-bench/apps/frappe/frappe/app.py\", line 64, in application
    response = frappe.api.handle()
  File \"/home/frappe/frappe-bench/apps/frappe/frappe/api.py\", line 58, in handle
    return frappe.handler.handle()
  File \"/home/frappe/frappe-bench/apps/frappe/frappe/handler.py\", line 30, in handle
    data = execute_cmd(cmd)
  File \"/home/frappe/frappe-bench/apps/frappe/frappe/handler.py\", line 69, in execute_cmd
    return frappe.call(method, **frappe.form_dict)
  File \"/home/frappe/frappe-bench/apps/frappe/frappe/__init__.py\", line 1086, in call
    return fn(*args, **newargs)
  File \"/home/frappe/frappe-bench/apps/frappe/frappe/handler.py\", line 97, in runserverobj
    frappe.desk.form.run_method.runserverobj(method, docs=docs, dt=dt, dn=dn, arg=arg, args=args)
  File \"/home/frappe/frappe-bench/apps/frappe/frappe/desk/form/run_method.py\", line 46, in runserverobj
    r = doc.run_method(method, args)
  File \"/home/frappe/frappe-bench/apps/frappe/frappe/model/document.py\", line 827, in run_method
    out = Document.hook(fn)(self, *args, **kwargs)
  File \"/home/frappe/frappe-bench/apps/frappe/frappe/model/document.py\", line 1119, in composer
    return composed(self, method, *args, **kwargs)
  File \"/home/frappe/frappe-bench/apps/frappe/frappe/model/document.py\", line 1102, in runner
    add_to_return_value(self, fn(self, *args, **kwargs))
  File \"/home/frappe/frappe-bench/apps/frappe/frappe/model/document.py\", line 821, in <lambda>
    fn = lambda self, *args, **kwargs: getattr(self, method)(*args, **kwargs)
  File \"/home/frappe/frappe-bench/apps/erpnext/erpnext/accounts/doctype/pos_invoice/pos_invoice.py\", line 287, in set_missing_values
    pos = self.set_pos_fields(for_validate)
  File \"/home/frappe/frappe-bench/apps/erpnext/erpnext/accounts/doctype/pos_invoice/pos_invoice.py\", line 228, in set_pos_fields
    update_multi_mode_option(self, pos)
  File \"/home/frappe/frappe-bench/apps/erpnext/erpnext/accounts/doctype/sales_invoice/sales_invoice.py\", line 1624, in update_multi_mode_option
    payment_mode[0].default = pos_payment_method.default
IndexError: list index out of range
```

**Comment by olragon** [comment]
I managed to fix these issues:

1.` Unknown column 'display_items_in_stock' in 'field list'`:
It's already fixed by 8aeb20cab8e7e6f2c88bece600579aad8c5cb6e5, temporary solution for 13.0.0-beta.4 is using Form Customization create field `display_items_in_stock` with type = Check for Open Point of Sale Profile.
2. `payment_mode[0].default = pos_payment_method.default IndexError: list index out of range`
We need to set Default Accounts for all enabled Payment method.
Accounting -> Settings: Mode of Payment -> Add Default Accounts for all enabled Payment method.

**Comment by PatroxGaurab** [comment]
That worked.. Many thanks for the fix.
